rewrite-gradle/src/main/java/org/openrewrite/gradle/UpgradeDependencyVersion.java

```diff
@@ -262,7 +262,7 @@ public J visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx)
                                         acc.versionPropNameToGA.put(versionVariableName, ga);
                                         acc.gaToNewVersion.put(ga, resolvedVersion);
                                     }
-                                } catch (MavenDownloadingException e) {
+                                } catch (IllegalStateException | MavenDownloadingException e) {
```

Comment:
This probably should instead be fixed here:
https://github.com/openrewrite/rewrite/blob/main/rewrite-gradle%2Fsrc%2Fmain%2Fjava%2Forg%2Fopenrewrite%2Fgradle%2FDependencyVersionSelector.java#L168

The rationale is that the version is being compared and should really be skipped as exotic. The `DependencyVersionSelector` is used in several spots all of which _could_ encounter this exact same issue. By moving this fix to the location in the link, we'll fix all of the recipes in one fell swoop.

Reply:
I did consider that also. However, I thought "If you use the versionSelector, you **know** that  a version is selectable".

What would you return if no version could be selected?

Reply:
@shanman190 moved the check to the proposed position!

Reply:
Sweet!

Yeah, the version selector is simply trying to answer the question of: "is there a version to which we can upgrade to?" At which point, the version selector should return only the value to be upgraded to or null to indicate that there was no upgrade available or possible. In particular, with versions like the one from the issue, there's less of a natural order (outside of ASCII) in order to ensure that we are moving up the version strictly. This appears as well with the old Spring Cloud release trains where they were using the alpha ordering of train stations across Europe to order their releases (ie. Angel, Brixton, Camden, etc).